processing/src/main/java/io/druid/query/select/SelectQueryQueryToolChest.java

```diff
@@ -243,29 +266,44 @@ public Object apply(final Result<SelectResultValue> input)
 
             DateTime timestamp = granularity.toDateTime(((Number) resultIter.next()).longValue());
 
-            return new Result<SelectResultValue>(
+            List<DimensionSpec> dimensionSpecs = query.getDimensions();
+
+            Map<String, Integer> pageIdentifier = jsonMapper.convertValue(
+                resultIter.next(), new TypeReference<Map<String, Integer>>() {}
+                );
+            Set<String> dimensionSet = jsonMapper.convertValue(
+                resultIter.next(), new TypeReference<Set<String>>() {}
+            );
+            Set<String> metricSet = jsonMapper.convertValue(
+                resultIter.next(), new TypeReference<Set<String>>() {}
+            );
+            List<EventHolder> eventHolders = jsonMapper.convertValue(
+                resultIter.next(), new TypeReference<List<EventHolder>>() {}
+                );
+            if (dimensionSpecs != null) {
```

Comment:
The query engine treats empty list and null the same (both mean "include everything as a default dimension") so maybe this should be `if (dimensionSpecs != null && !dimensionSpecs.isEmpty())`